src/uu/pidwait/src/pidwait.rs

```diff
@@ -0,0 +1,246 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use clap::{arg, crate_version, Arg, ArgAction, ArgMatches, Command};
+use regex::Regex;
+use std::{collections::HashSet, env, sync::OnceLock};
+use uu_pgrep::process::{walk_process, ProcessInformation, RunState, Teletype};
+use uucore::{
+    error::{UResult, USimpleError},
+    format_usage, help_about, help_usage,
+};
+use wait::waiting;
+
+mod wait;
+
+const ABOUT: &str = help_about!("pidwait.md");
+const USAGE: &str = help_usage!("pidwait.md");
+
+static REGEX: OnceLock<Regex> = OnceLock::new();
+
+#[derive(Debug)]
+struct Settings {
+    echo: bool,
+    count: bool,
+    full: bool,
+    ignore_case: bool,
+    newest: bool,
+    oldest: bool,
+    older: Option<u64>,
+    terminal: Option<HashSet<Teletype>>,
+    exact: bool,
+    runstates: Option<RunState>,
```

Comment:
I think this is not correct because you can specify more than one `RunState`: `pidwait --runstates=D,S`